lms/djangoapps/course_api/views.py

```diff
@@ -2,52 +2,122 @@
 Course API Views
 """
 
-from rest_framework.views import APIView
-from rest_framework.response import Response
-from rest_framework.reverse import reverse
+from rest_framework.exceptions import NotFound, PermissionDenied
+from rest_framework.views import APIView, Response
 
+from opaque_keys import InvalidKeyError
 from opaque_keys.edx.keys import CourseKey
+
+
 from openedx.core.lib.api.view_utils import view_auth_classes
-from xmodule.modulestore.django import modulestore
+
+from .api import course_detail, list_courses
 
 
 @view_auth_classes()
-class CourseView(APIView):
+class CourseDetailView(APIView):
     """
-    Course API view
+    View class to describe a course
     """
 
     def get(self, request, course_key_string):
         """
         Request information on a course specified by `course_key_string`.
-            Body consists of a `blocks_url` that can be used to fetch the
-            blocks for the requested course.
+
+        Body consists of the following fields:
+            * `blocks_url`: can be used to fetch the blocks
+            * `course_image`:
+            * `description`:
+            * `end`: Date the course ends
+            * `enrollment_end`: Date enrollment ends
+            * `enrollment_start`: Date enrollment begins
+            * `id`: Course key
+            * `name`: Name of the course
+            * `number`: Catalog number of the course
+            * `org`: Name of the organization that owns the course
+            * `start`: Date the course begins
+            * `start_display`: Readably formatted start of the course
+            * `start_type`: Hint describing how `start_display` is set. One of:
+                  - `"string"`: manually set
+                  - `"timestamp"`: generated form `start` timestamp
+                  - `"empty"`: the start date should not be shown
 
         Arguments:
-            request (HttpRequest)
-            course_key_string
+            request: (HttpRequest)
+            course_key_string: (string) Key for the desired course
+
+        Parameters:
+            username (optional):
+                The username of the specified user whose visible courses we
+                want to see.  Defaults to the current user.
 
         Returns:
             HttpResponse: 200 on success
 
-
         Example Usage:
-
             GET /api/courses/v1/[course_key_string]
             200 OK
 
         Example response:
+            {
+                "blocks_url": "/api/courses/v1/blocks/?course_id=edX%2Ftoy%2F2012_Fall",
```

Comment:
@nasthagiri @jcdyer I've had the same challenge with APIs that switch from paginated to not. I believe that the recommendation was to use the current format because that's what DRF handles. Here's the current convention:

https://openedx.atlassian.net/wiki/display/AC/edX+REST+API+Conventions#edXRESTAPIConventions-7.Pagination

Note that paginated results do need to be treated very differently so in some ways it is okay if the formats aren't the same. My general feeling is that it makes sense to default all list APIs to be paginated unless there's a strong reason not to, so there's not much switching of APIs between paginated and not.

Having said all that, if we were starting again I'd like to do something like you guys are thinking. I recommend mentioning it in the API room and getting the pulse of other interested folks. It would be quite a large change to switch over all the existing paginated APIs because the change is incompatible but it might make sense to do it anyway before we build too many more. A lot of our web UIs are using the shared Backbone pagination components that T&L built, so that insulates a lot of the code from knowing the details of the format. It would be a good opportunity to get them all switched over.

Note that @peter-fogg and I are working on moving these paginated components into the UI Toolkit so that they can be used by other repositories too:

https://github.com/edx/edx-ui-toolkit/pull/2